src/network/networkd-sysctl.c

```diff
@@ -5,17 +5,189 @@
 #include <linux/if_arp.h>
 
 #include "af-list.h"
+#include "cgroup-util.h"
+#include "fd-util.h"
 #include "missing_network.h"
 #include "networkd-link.h"
 #include "networkd-lldp-tx.h"
 #include "networkd-manager.h"
 #include "networkd-ndisc.h"
 #include "networkd-network.h"
 #include "networkd-sysctl.h"
+#include "path-util.h"
 #include "socket-util.h"
 #include "string-table.h"
 #include "sysctl-util.h"
 
+#if HAVE_VMLINUX_H
+
+#include "bpf-link.h"
+
+#include "bpf/sysctl_monitor/sysctl-monitor-skel.h"
+#include "bpf/sysctl_monitor/sysctl-write-event.h"
+
+static struct sysctl_monitor_bpf *sysctl_monitor_bpf_free(struct sysctl_monitor_bpf *obj) {
+        sysctl_monitor_bpf__destroy(obj);
+        return NULL;
+}
+
+static struct ring_buffer *rb_free(struct ring_buffer *rb) {
+        sym_ring_buffer__free(rb);
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct sysctl_monitor_bpf *, sysctl_monitor_bpf_free);
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct ring_buffer *, rb_free);
+
+static int sysct_write_event(void *ctx, void *data, size_t data_sz) {
+        struct sysctl_write_event *we = ASSERT_PTR(data);
+        Hashmap **sysctl_shadow = ASSERT_PTR(ctx);
+        _cleanup_free_ char *path = NULL;
+        char *value;
+
+        path = path_join("/proc/sys", we->name);
+        if (!path)
+                return log_oom();
+
+        /* If we never managed this handle, ignore it. */
+        value = hashmap_get(*sysctl_shadow, path);
+        if (!value)
+                return 0;
+
+        if (!streq(value, we->newvalue))
+                log_warning("'%s' changed sysctl '%s' from '%s' to '%s', we want '%s'", we->comm, we->name, we->current, we->newvalue, value);
+
+        return 0;
+}
+
+static int on_ringbuf_io(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
+        struct ring_buffer *rb = ASSERT_PTR(userdata);
+        int r;
+
+        r = sym_ring_buffer__poll(rb, 1000);
```

Comment:
hmm, the 2nd arg is milliseconds of blocking? what's the rationale here? we run an even loop, it's not ok to block that long? shouldnÄ't this be zero? what's the rationale here?

Reply:
also, we typically want that "opaque" parameters that give us no hint what they are (and in this case even: in which unit) are prefixed with a parameter name as comment, i.e. here it could be:

```
r = sym_ring_buffer__poll(rb, /* timeout_msec= */ 1000);
```

or so.

(but again, it shouldn't be 1000 anyway, but zero, iiuc)